debezium-connector-oracle/src/main/java/io/debezium/connector/oracle/jsqlparser/SimpleDmlParser.java

```diff
@@ -0,0 +1,287 @@
+/*
+ * Copyright Debezium Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.debezium.connector.oracle.jsqlparser;
+
+import java.io.StringReader;
+import java.util.Collections;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import io.debezium.connector.oracle.antlr.listener.ParserUtils;
+import io.debezium.connector.oracle.logminer.OracleChangeRecordValueConverter;
+import io.debezium.connector.oracle.logminer.valueholder.LogMinerColumnValue;
+import io.debezium.connector.oracle.logminer.valueholder.LogMinerColumnValueImpl;
+import io.debezium.connector.oracle.logminer.valueholder.LogMinerColumnValueWrapper;
+import io.debezium.connector.oracle.logminer.valueholder.LogMinerDmlEntry;
+import io.debezium.connector.oracle.logminer.valueholder.LogMinerDmlEntryImpl;
+import io.debezium.data.Envelope;
+import io.debezium.relational.Column;
+import io.debezium.relational.Table;
+import io.debezium.relational.TableId;
+import io.debezium.relational.Tables;
+import io.debezium.text.ParsingException;
+
+import net.sf.jsqlparser.JSQLParserException;
+import net.sf.jsqlparser.expression.Alias;
+import net.sf.jsqlparser.expression.Expression;
+import net.sf.jsqlparser.expression.ExpressionVisitorAdapter;
+import net.sf.jsqlparser.expression.operators.relational.EqualsTo;
+import net.sf.jsqlparser.expression.operators.relational.ExpressionList;
+import net.sf.jsqlparser.expression.operators.relational.IsNullExpression;
+import net.sf.jsqlparser.expression.operators.relational.ItemsList;
+import net.sf.jsqlparser.expression.operators.relational.ItemsListVisitorAdapter;
+import net.sf.jsqlparser.parser.CCJSqlParserManager;
+import net.sf.jsqlparser.statement.Statement;
+import net.sf.jsqlparser.statement.delete.Delete;
+import net.sf.jsqlparser.statement.insert.Insert;
+import net.sf.jsqlparser.statement.update.Update;
+
+/**
+ * This class does parsing of simple DML: insert, update, delete.
+ * Log Miner supplies very simple syntax , that this parser should be sufficient to parse those.
+ * It does no support joins, merge, sub-selects and other complicated cases, which should be OK for Log Miner case
+ */
+public class SimpleDmlParser {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(SimpleDmlParser.class);
+    protected String catalogName;
```

Comment:
Can be made some final? Also this could be using a `TableId` here?

Reply:
Looking at the code here I think honestly that `table` could be factored out of being a class member and instead simply be an argument that is passed to the methods that actually use it instead; however I'd rather do this as a code-cleanup task later because I'm not sure with the new parser implementation from Andrey & team, how much of this code would remain.

I've made the instance variable changes to those that can be final; and I logged a refactor task [DBZ-2532](https://issues.redhat.com/browse/DBZ-2532).